tools/hicup/hicup_hicup.xml

```diff
@@ -0,0 +1,108 @@
+<tool id="hicup_hicup" name="Pipeline" version="0.1.0">
+    <requirements>
+        <requirement type="package" version="0.5.9">hicup</requirement>
+        <requirement type="package" version="2.2.6">bowtie2</requirement>
+        <requirement type="package" version="1.2">samtools</requirement>
+    </requirements>
+    <stdio>
+        <exit_code range="1:" />
+    </stdio>
+    <command><![CDATA[
+        BOWTIE_PATH_BASH="\$(which bowtie2)" &&
+        #set index_path = ''
+        #if str($reference_genome.source) == "history":
+            bowtie2-build "$reference_genome.own_file" genome &amp;&amp;
+            ln -s "$reference_genome.own_file" genome.fa &amp;&amp;
+            #set index_path = 'genome'
+        #else:
+            #set index_path = $reference_genome.index.fields.path
+        #end if
+        hicup 
+        --digest $digest 
+        --index $index_path 
+        --bowtie2 \$BOWTIE_PATH_BASH
+        $keep $temp $zip $nofill
+        
+        #if $format
+            --format $format
+        #end if    
+        #if $threads:
+            --threads $threads
+        #end if
+        #if $longest:
+            --longest $longest
+        #end if
+        #for $file_pair in $input_files:
+            $file_pair.input_first_sequence $file_pair.input_second_sequence
+        #end for
+        && mv hicup_filter_ditag_rejects* hicup_filter_ditag_rejects
+    ]]></command>
+    <inputs>
+        <repeat name="input_files" min="1" max="10000" title="Select input file(s):">
```

Comment:
1000?
You want to use a collection here isn't it? Please look at other tools that do similar things, bowtie2 ...